RecipeApp.Core/RecipeConverter.cs

```diff
@@ -0,0 +1,270 @@
+﻿using RecipeApp.Core.Exceptions;
+using RecipeApp.Core.Units;
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace RecipeApp
+{
+    public class RecipeConverter
+    {
+        public static string ConvertRecipe(string recipe, bool isSiUnit)
+        {
+            if (isSiUnit)
+            {
+                return ConvertToSiUnits(recipe);
+            }
+            else
+            {
+                return ConvertToCookingUnits(recipe);
+            }
+        }
+
+        public static string ConvertToSiUnits(string recipe)
+        {
+            string[] words = recipe.Split(' ', '\n');
+
+            for (int index = 0; index < words.Length; index++)
+            {
+                try
+                {
+                    StandardiseCookingUnit(index, words);
+                }
+                catch (InvalidRecipeException ex)
+                {
+                    Console.WriteLine($"Skipping word, because: {ex.Message}");
+                }
+            }
+
+            return string.Join(" ", words);
+        }
+
+        //public static string MakeUnitsPretty(string recipe)
+        //{
+        //    string[] words = recipe.Split(' ', '\n');
+
+        //    for (int index = 0; index < words.Length; index++)
+        //    {
+        //        if (words[index] == "ml" && ParseDouble(words[index - 1], out double amount))
+        //        {
+        //            if (amount > 100)
+        //            {
+        //                words[index] = "l";
+        //                words[index - 1] = (amount / 100).ToString("F2");
+        //            }
+        //        }
+
+        //        var cookingUnit = words[index];
+        //        var multiplier = FindSiMultiplier(cookingUnit);
+        //        if (multiplier == -1) continue;
+
+        //        ParseDouble(words[index - 1], out amount);
+
+        //        words[index] = GetClosestCookingUnit(amount);
+        //        words[index - 1] = (amount * multiplier / Conversion.CookingUnits[words[index]]).ToString("F2");
+        //    }
+
+        //    return string.Join(" ", words);
+        //}
+
+        public static string MakeRecipePretty(string words)
```

Comment:
The name of a function is not clear. Instead of pretty, state what you do with a recipe. If it's not possible to express that with a reasonably short method name, use a comment.

Reply:
The length is irrelevant. But it should do one thing. So, if you ever consider splitting a method- split it. Don't force the reader to read through all the code (regardless how big it is) if all you want to understand is the big picture. If you want to know the details, you will go deeper into private methods.

Method doing 1 thing- means 1 level of abstraction. So for example if you call a method SendEmail and next to that you run a loop- loop is a low level and thus should be moved to a method of its own.

Reply:
Ok , got it ^^